File: leetcode/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp, s = [0]*n, 1
        dp[n-1] = 1
        for i in range(n-2, -1, -1):
            if nums[i] >= s:
                dp[i], s = 1, 1
            else:
                s += 1
                continue
        return True if dp[0] == 1 else False

File: leetcode/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        vector<int> dp;
        int n = nums.size();
        for (auto i = 0; i < n; i++) {
            dp.push_back(0);
        } 
        dp[n-1] = 1; int s = 1;
        for (auto i = n-2; i >= 0; i--) {
            if (nums[i] >= s) {s = 1; dp[i] = 1;}
            else s += 1;
        }
        return dp[0];
    }

};